app/start_handlers.py

```diff
@@ -47,41 +47,42 @@ async def post(self, request):
 
         data = await request.post()
 
-        if (not data.get('uac')) or (data.get('uac') == ''):
+        uac = data.get('uac')
+        if not uac:
             return self._display_missing_uac_error(request, display_region)
 
+        combined_uac = uac.upper().replace(' ', '')
+        if len(combined_uac) < UAC_LENGTH:
+            return self._display_malformed_uac_message(request, display_region, BAD_CODE_LENGTH_MSG,
+                                                       BAD_CODE_LENGTH_MSG_CY)
+
         try:
-            request['uac_hash'] = self._uac_hash(data.get('uac'))
+            request['uac_hash'] = self._uac_hash(combined_uac)
         except TypeError:
-            return self._display_malformed_uac_message(request, display_region)
+            return self._display_malformed_uac_message(request, display_region, INVALID_CODE_MSG, INVALID_CODE_MSG_CY)
 
         token = await EqLaunch.get_token(request, display_region, request.app)
         EqLaunch.call_eq(request.app['EQ_URL'], token)
 
     @staticmethod
-    def _display_malformed_uac_message(request, display_region):
+    def _display_malformed_uac_message(request, display_region, en_code, cy_code):
         logger.warn('attempt to use a malformed access code',
                     client_ip=request['client_ip'], client_id=request['client_id'], trace=request['trace'])
         message = {
-            'en': INVALID_CODE_MSG,
-            'cy': INVALID_CODE_MSG_CY
+            'en': en_code,
+            'cy': cy_code
         }[display_region]
         flash(request, message)
         return HTTPFound(request.app.router['Start:get'].url_for(display_region=display_region))
 
     @staticmethod
-    def _uac_hash(uac, expected_length=16):
-        if uac:
-            combined = uac.upper().replace(' ', '')
-        else:
-            combined = ''
-
-        uac_validation_pattern = re.compile(r'^[A-Z0-9]{16}$')
+    def _uac_hash(uac):
+        uac_validation_pattern = re.compile(fr'^[A-Z0-9]{{{UAC_LENGTH}}}$')   # Outer 2 curly braces escape the f-string
 
-        if (len(combined) < expected_length) or not (uac_validation_pattern.fullmatch(combined)):  # yapf: disable
```

Comment:
The first part of this statement has been moved to the calling function, as we want to let the user know if the length is under the expected UAC length